tools/bridge/bridge/confirmation_task_planner.py

```diff
@@ -1,161 +1,88 @@
-            if timestamped_transfer_hash.transfer_hash != transfer_hash
-        ]
-
-    def forget_old_transfer_hashes(self, cutoff_timestamp: int) -> None:
-        """Drop all transfer hashes that happened before the given cutoff timestamp."""
-        self.timestamped_transfer_hashes = [
-            timestamped_transfer_hash
-            for timestamped_transfer_hash in self.timestamped_transfer_hashes
-            if timestamped_transfer_hash.timestamp >= cutoff_timestamp
-        ]
-
-    def transfer_hash_exists(self, transfer_hash: Hash32) -> bool:
-        """Check if a transfer hash has been recorded and not yet forgotten or not."""
-        return any(
-            timestamped_transfer_hash.transfer_hash == transfer_hash
-            for timestamped_transfer_hash in self.timestamped_transfer_hashes
-        )
-
-    def get_transfer_hashes(self, start_time: int, end_time: int) -> List[Hash32]:
-        """Return all transfer hashes in a given (inclusive) time interval."""
-        return [
-            timestamped_transfer_hash.transfer_hash
-            for timestamped_transfer_hash in self.timestamped_transfer_hashes
-            if start_time <= timestamped_transfer_hash.timestamp <= end_time
-        ]
-
-
-class ConfirmationTaskPlanner:
-    def __init__(self, home_look_ahead: int, history_length: int) -> None:
-        self.transfer_event_recorder = TransferHashRecorder()
-        self.confirmation_event_recorder = TransferHashRecorder()
-        self.completion_event_recorder = TransferHashRecorder()
-
-        self.recently_processed_transfer_hashes: Set[Hash32] = set()
-
-        self.home_look_ahead = home_look_ahead
-        self.history_length = history_length
-
-        self.transfers_processed_until = 0
-
-    def apply_transfer_hash(
-        self, event: str, timestamp: int, transfer_hash: Optional[Hash32]
-    ) -> None:
-        if event == "Transfer":
-            recorder = self.transfer_event_recorder
-        elif event == "Confirmation":
-            recorder = self.confirmation_event_recorder
-        elif event == "Completion":
-            recorder = self.completion_event_recorder
+from eth_utils import decode_hex
+from web3.datastructures import AttributeDict
+
+from bridge.constants import (
+    COMPLETION_EVENT_NAME,
+    CONFIRMATION_EVENT_NAME,
+    TRANSFER_EVENT_NAME,
+)
+
+
+class TransferRecorder:
+    def __init__(self, sync_persistence_time: float) -> None:
+        self.sync_persistence_time = sync_persistence_time
+
+        self.transfer_events: Dict[Hash32, AttributeDict] = {}
+
+        self.transfer_hashes: Set[Hash32] = set()
+        self.confirmation_hashes: Set[Hash32] = set()
+        self.completion_hashes: Set[Hash32] = set()
+
+        self.scheduled_hashes: Set[Hash32] = set()
+
+        self.confirmations_synced_until = 0.0
+        self.completions_synced_until = 0.0
+
+    def apply_sync_completed(self, event: str, timestamp: float) -> None:
```

Comment:
According to the following `apply_event()` function, I would go for `event_name` here. Because in the following an `event` is an actual `dict` with all the attributes and the `event_name` must be extracted before the evaluation.

```suggestion
    def apply_sync_completed(self, event_name: str, timestamp: float) -> None:
```